nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBRepository.java

```diff
@@ -1126,15 +1128,20 @@ public void clearLockManagerCaches() {
     }
 
     protected FindIterable<Document> find(Bson filter) {
-        return coll.find(filter).maxTime(maxTimeMS, MILLISECONDS);
+        return coll.find(filter).maxTime(getMaxTimeMs(), MILLISECONDS);
+    }
+
+    protected long getMaxTimeMs() {
+        long ttl = TransactionHelper.getTransactionTimeToLive();
+        return ttl > 0 ? ttl * 1000 : maxTimeMS;
```

Comment:
Ok I missed it on master but this means that if the transaction has just timed out (ttl=0), we'll use the full max transaction time? I'm not sure that's ok...

Reply:
ok I will add a fix commit on master & lts